packages/core/src/package/componentFilter/EntitlementVersionFilter.ts

```diff
@@ -0,0 +1,78 @@
+import { ComponentSet, registry } from '@salesforce/source-deploy-retrieve';
+import SFPOrg from '../../org/SFPOrg';
+import QueryHelper from '../../queryHelper/QueryHelper';
+import SFPLogger, { Logger, LoggerLevel } from '../../logger/SFPLogger';
+import { DeploymentFilter } from './DeploymentFilter';
+import * as fs from 'fs-extra';
+const { XMLBuilder } = require('fast-xml-parser');
+
+const EXISTING_SLAPPROCESS_QUERY = `SELECT Name, NameNorm,VersionNumber, VersionMaster FROM SlaProcess ORDER BY VersionNumber DESC`;
+
+export default class EntitlementVersionFilter implements DeploymentFilter {
+    public async apply(org: SFPOrg, componentSet: ComponentSet,logger:Logger): Promise<ComponentSet> {
+        //Only do if entitlment exits in the package
+        SFPLogger.log(`Filtering Entitlement Process....`,LoggerLevel.INFO,logger);
+        let sourceComponents = componentSet.getSourceComponents().toArray();
+        let isEntitlementFound: boolean = false;
+        for (const sourceComponent of sourceComponents) {
+            if (sourceComponent.type.name === registry.types.entitlementprocess.name) {
+                isEntitlementFound = true;
+                break;
```

Comment:
should we get a list of all entitlement processes here, in case the component set is very large

Reply:
I dont think there is anything we can do..  even the normal one would be a filter, Let me check